packages/field-base/src/slot-styles-mixin.js

```diff
@@ -0,0 +1,88 @@
+import { dedupingMixin } from '@polymer/polymer/lib/utils/mixin.js';
+
+const stylesMap = new WeakMap();
+
+/**
+ * Get all the styles inserted into root.
+ * @param {DocumentOrShadowRoot} root
+ * @return {Array<string>}
+ */
+function getRootStyles(root) {
+  if (!stylesMap.has(root)) {
+    stylesMap.set(root, []);
+  }
+
+  return stylesMap.get(root);
+}
+
+/**
+ * Detect if styles have been already inserted to root.
+ * @param {string} styles
+ * @param {DocumentOrShadowRoot} root
+ * @return {boolean}
+ */
+function hasStyles(styles, root) {
+  return getRootStyles(root).includes(styles);
+}
+
+/**
+ * Insert styles into the root.
+ * @param {string} css
+ * @param {DocumentOrShadowRoot} root
+ */
+function insertStyles(styles, root) {
+  if (!styles || (styles && hasStyles(styles, root))) {
+    return;
+  }
+
+  const rootNode = root === document ? document.head : root;
+  const style = document.createElement('style');
+  style.appendChild(document.createTextNode(styles));
+  rootNode.appendChild(style);
+}
+
+const SlotStylesMixinImplementation = (superclass) =>
+  class SlotStylesMixinClass extends superclass {
+    /**
+     * List of styles to insert into root.
+     * @protected
+     */
+    get slotStyles() {
+      return {};
+    }
+
+    /** @protected */
+    connectedCallback() {
+      super.connectedCallback();
+
+      this.__applySlotStyles();
```

Comment:
It's a bit unclean that if you have 10 number fields on a page, you'll also have 10 identical `<style>` tags for styling the slotted input.

Reply:
Fixed.